featurebyte/service/target_helper/base_feature_or_target_computer.py

```diff
@@ -191,3 +179,84 @@ async def compute(
             validation_parameters=validation_parameters,
         )
         await self.query_executor.execute(params)
+
+    async def _compute_data_frame(
+        self,
+        observation_set: pd.DataFrame,
+        compute_request: ComputeRequestT,
+        output_table_details: TableDetails,
+    ) -> None:
+        validation_parameters = await self.get_validation_parameters(compute_request)
+        request_column_names = set(observation_set.columns)
+        await self._execute(
+            observation_set=observation_set,
+            compute_request=compute_request,
+            output_table_details=output_table_details,
+            validation_parameters=validation_parameters,
+            request_column_names=request_column_names,
+        )
+
+    async def _compute_observation_table(
+        self,
+        observation_set: ObservationTableModel,
+        compute_request: ComputeRequestT,
+        output_table_details: TableDetails,
+    ) -> None:
+        validation_parameters = await self.get_validation_parameters(compute_request)
+        request_column_names = {col.name for col in observation_set.columns_info}
+
+        feature_cache, nodes_hash_dict = await self.feature_table_cache_service.filter_nodes(
+            observation_set.id,
+            validation_parameters.graph,
+            [node.name for node in validation_parameters.nodes],
+        )
+        non_cached_validation_parameters = replace(
+            validation_parameters, nodes=list(nodes_hash_dict.values())
+        )
+
+        if len(nodes_hash_dict) < len(validation_parameters.nodes):
+            table_details = output_table_details.copy()
+            table_details.table_name = (
+                f"TEMP_{MaterializedTableNamePrefix.FEATURE_TABLE_CACHE}_{ObjectId()}"
+            )
+        else:
+            table_details = output_table_details.copy()
+            table_details.table_name = feature_cache.table_name
```

Comment:
We cannot let `self._execute()` create the feature cache table because the table created by `self._execute()` will have feature names as the column names, but the column names in the feature cache table should be based on definition hash. We will need to manage the feature cache table creation separately.

Reply:
Where does it get the columns names? From nodes? 
Do I need to modify Nodes to change column names? 

This is the part which is very cryptic to me - it is quite hard to understand how some code is translated into the SQL query. 
To me it looks like a set of dataclasses/pydantic models which is then magically executed somehow.